mp4parse/src/lib.rs

```diff
@@ -752,6 +752,10 @@ pub struct MetadataBox {
     pub sort_album_artist: Option<TryString>,
     /// The name of the composer to sort by 'soco'
     pub sort_composer: Option<TryString>,
+    /// XML metadata
+    pub xml: Option<TryString>,
+    /// Binary XML metadata
+    pub bxml: Option<TryVec<u8>>,
```

Comment:
Based on ISOBMFF (ISO/IEC 14496-12:2015) § 8.11.2.1 (emphasis mine):
> Quantity: Zero or one
…
> When the primary data is in XML format and it is desired that the XML be stored directly in the meta‐ box, **one of these forms may be used**.

I think making it structurally clear that only one or the other can be used is helpful:

```suggestion
    // metadata
    pub xml: Option<XmlBox>,
}

/// See ISOBMFF (ISO 14496-12:2015) § 8.11.2.1
#[derive(Debug)]
pub enum XmlBox {
    /// XML metadata
    StringXmlBox(TryString),
    /// Binary XML metadata
    BinaryXmlBox(TryVec<u8>),
}

```